src/renderers/dom/ReactDOMNodeStreamRenderer.js

```diff
@@ -19,10 +19,8 @@ var Readable = require('stream').Readable;
 
 // This is a Readable Node.js stream which wraps the ReactDOMPartialRenderer.
 class ReactMarkupReadableStream extends Readable {
-  constructor(element, makeStaticMarkup) {
-    // Calls the stream.Readable(options) constructor. Consider exposing built-in
-    // features like highWaterMark in the future.
-    super({});
+  constructor(element, makeStaticMarkup, {highWaterMark} = {}) {
```

Comment:
Great question!

The reason I'm destructing `highWaterMark` is that there are several other options that can be passed to the `Readable` constructor (documented [here](https://nodejs.org/api/stream.html#stream_new_stream_readable_options)), and those options will make the stream behave incorrectly. For example, if you pass an `encoding` argument to the `Readable` constructor, the stream will output gibberish. Passing a `read` argument will make the stream not render anything.

If we pass the `options` object through unfiltered, the client of this class could accidentally make it malfunction, whereas if we only pass `highWaterMark`, we will be OK. Does that make sense?